torch/distributed/fsdp/_symbolic_trace.py

```diff
@@ -0,0 +1,220 @@
+import contextlib
+import functools
+from dataclasses import dataclass, field
+from typing import Any, Callable, Dict, Generator, List, Optional, Set, Tuple
+
+import torch
+
+
+__all__ = ["TracingConfig"]
+
+
+@dataclass
+class TracingConfig:
+    """
+    Configurations used in ``ParamExecOrderWrapPolicy`` for symbolic tracing of a model.
+
+    Args:
+        tracer (torch.fx.Tracer): An instance of ``torch.fx.Tracer`` that will be used to perform symbolic
+            tracing. ``tracer`` is default to be ``torch.fx.Tracer()``, but can also be instance
+            of some child class of ``torch.fx.Tracer``. For example, one may want to use ``HFTracer``
+            for models in Transformers:
+            .. _Transformers: https://huggingface.co/docs/transformers/index
+
```

Comment:
nit: Do these newlines affect rendering? (I think we cannot really tell here since we did not include this file in the `index.rst`.)

Reply:
yeah I'm actually not sure, but removed the space in case that has an effect